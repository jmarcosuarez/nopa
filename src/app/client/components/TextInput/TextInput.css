.errorPlaceholder {

}
.error {
  background: red;
}

.formInput {
  margin: 20px;

  label {
    display: none;
  }

  input {
    padding: 10px;
    min-width: 300px;
    border: 0;
    border-radius:1px;
    border-bottom: 3px solid lightgray;
  }
}
