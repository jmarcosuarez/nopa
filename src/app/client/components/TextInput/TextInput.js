/* eslint-disable no-confusing-arrow */
import React from 'react';
import { compose, head } from 'ramda';
import { isValid } from '../../containers/Login/Revalidation';
import styles from './TextInput.css';

const createErrorMessage = 
  errorMsgs => isValid(errorMsgs) ? null : <div className={styles.error}>{head(errorMsgs)}</div>;

const TextInput = props => (
  <div className={styles.formInput}>
    <label htmlFor={props.name}>{props.name}</label>
    <input
      id={props.name}
      name={props.name}
      type="text"
      className={isValid(props.errors) ? '' : 'error'}
      value={props.form}
      onChange={props.onChange}
    />
    <div className={styles.errorPlaceholder}>{ createErrorMessage(props.errors) }</div>
  </div>
  );

TextInput.propTypes = {
  name: React.PropTypes.string.isRequired,
  form: React.PropTypes.object.isRequired,
  onChange: React.PropTypes.func.isRequired,
  errors: React.PropTypes.array,
};

TextInput.defaultProps = {
  errors: [],
};


export default TextInput;
