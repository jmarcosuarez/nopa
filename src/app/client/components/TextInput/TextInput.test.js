/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TextInput from './TextInput';

test('should render correctly', () => {
  const fc = jest.fn();
  const object = {
    surname: 'Denis', 
    sortCode: '23-43-54',
    accountNumber: '09434534343434',
    passCode: '0943',
    memorableWord: 'Qwteyruy',
  };
  const wrapper = shallow(
    <TextInput
      name={object.surname}
      value={object.surname}
      onChange={fc} onErrors={fc}
    />,
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
