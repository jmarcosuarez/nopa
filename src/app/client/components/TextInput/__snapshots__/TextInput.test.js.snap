// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="formInput"
>
  <label
    className="text-left"
    htmlFor="Denis"
  >
    Denis
  </label>
  <input
    className=""
    id="Denis"
    name="Denis"
    onChange={[Function]}
    type="text"
    value="Denis"
  />
  <div
    className="errorPlaceholder"
  />
</div>
`;
